akd/src/directory.rs

```diff
@@ -218,9 +212,7 @@ impl<S: Storage + Sync + Send, V: VRFKeyStorage> Directory<S, V> {
             if let Err(err) = self.storage.commit_transaction().await {
                 // ignore any rollback error(s)
                 let _ = self.storage.rollback_transaction().await;
-                return Err(AkdError::HistoryTreeNode(HistoryTreeNodeError::Storage(
-                    err,
-                )));
+                return Err(AkdError::Storage(err));
```

Comment:
nit: Maybe add the origin to the message? "Couldn't commit transaction in publish"